src/modules/flow/http-client/http-client.c

```diff
@@ -550,33 +619,43 @@ common_post_process(struct sol_flow_node *node, void *data,
 }
 
 static int
-boolean_process_token(struct sol_flow_node *node,
-    struct sol_json_token *key, struct sol_json_token *value)
+boolean_process_token(struct sol_flow_node *node, struct sol_buffer *buf)
 {
-    bool result;
+    bool result, found;
+    enum sol_json_loop_reason reason;
+    struct sol_json_scanner sub_scanner;
+    struct sol_json_token sub_key, sub_value, token;
 
-    if (sol_json_token_get_type(value) == SOL_JSON_TYPE_TRUE)
-        result = true;
-    else if (sol_json_token_get_type(value) == SOL_JSON_TYPE_FALSE)
-        result = false;
-    else
-        return -EINVAL;
+    sol_json_scanner_init(&sub_scanner, buf->data, buf->used);
+    found = false;
+
+    SOL_JSON_SCANNER_OBJECT_LOOP (&sub_scanner, &token, &sub_key, &sub_value, reason) {
+        if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&sub_key, "value")) {
```

Comment:
do we need `{"value": true}`? Why not just `true`? I'm not sure if that would be valid json for other clients doing json http requests, like browsers.